File: Day2/part2.py
file = "input.txt"

inputs = []
with open(file, "r") as open_file:
    while (line := open_file.readline()):
        inputs.append(
            line.strip().split()
        )

horizontal = 0
depth = 0
aim = 0

for input in inputs:
    input[1] = int(input[1])
    if input[0] == "down":
        aim += input[1]

    elif input[0] == "up":
        aim -= input[1]

    elif input[0] == "forward":
        horizontal += input[1]
        depth += aim*input[1]

print(horizontal*depth)

File: Java/2021/Day2/part2.java
import java.io.File;
import java.io.FileNotFoundException;
import java.util.*;

class Part2 {
  public static void main(String[] args) throws FileNotFoundException {
    ArrayList<String[]> data = readFile("input.txt");

    int horizontal = 0;
    int depth = 0;
    int aim = 0;

    for (String[] action : data) {
      String command = action[0];
      int steps = Integer.parseInt(action[1]);

      switch (command) {
        case "forward":
          horizontal += steps;
          depth += aim * steps;
          break;

        case "down":
          aim += steps;
          break;

        case "up":
        aim -= steps;
          break;
      }
    }

    System.out.println(horizontal*depth);
  }

  private static ArrayList<String[]> readFile(String path) throws FileNotFoundException {
    File myFile = new File(path);
    Scanner fileScanner = new Scanner(myFile);

    ArrayList<String[]> data = new ArrayList<String[]>();

    while (fileScanner.hasNextLine()) {
      String[] line = fileScanner.nextLine().trim().split(" ");

      data.add(line);
    }

    return data;
  }
}
